### PR TITLE
Connected two usecase to UI and DAO, RecipeView added CollectActionListener, revised userID to username

### PR DESCRIPTION
**Pull request by yuxizzz**
- resolved merging conflict (merge from main)
- minor edits of the program
- change userID to username in my use cases

### CODE REVIEW

File: src/data_access/FileFolderDataAccessObject.java [hunk 1]
```diff
@@ -131,4 +132,17 @@ public Folder get(String folderName) {
     public HashMap<Integer, Recipe> getrecipeMap(String foldername) {
```
Comment (shiqio3o) [inline src/data_access/FileFolderDataAccessObject.java:132]: this function I did some changes in my branch

File: src/data_access/FileFolderDataAccessObject.java [hunk 2]
```diff
@@ -128,7 +129,31 @@ public Folder get(String folderName) {
      * @param foldername
      * @return
      */
-    public HashMap<Integer, Recipe> getrecipeMap(String foldername) {
-        return folders.get(foldername).getRecipeMap();
+  // TODO change it to integer list<recipe information> use java doc to explain
+//    list[url, title]
+    public HashMap<Integer, ArrayList> getrecipeMap(String foldername) {
```
Comment (shiqio3o) [inline src/data_access/FileFolderDataAccessObject.java:134]: I changed this function to the newest version

File: src/use_case/collect_recipe/CollectRecipeInputData.java [hunk 3]
```diff
@@ -1,18 +1,17 @@
 package use_case.collect_recipe;
 
 public class CollectRecipeInputData {
```
Comment (shiqio3o) [inline src/use_case/collect_recipe/CollectRecipeInputData.java:3]: why do you add collect recipe in opened folder view model, the collect button should in the opened recipe view

File: src/data_access/FileFolderDataAccessObject.java (hunk 1 above)
Comment (yuxizzz) [reply src/data_access/FileFolderDataAccessObject.java]: hi, but you are looking at the outdated version. if you check out the newer commits, this should be fixed.

File: src/data_access/FileFolderDataAccessObject.java (hunk 2 above)
Comment (yuxizzz) [reply src/data_access/FileFolderDataAccessObject.java:134]: did you push that to the main? if not, use mine for now and you can merge your things to the main.

File: src/use_case/collect_recipe/CollectRecipeInputData.java (hunk 3 above)
Comment (yuxizzz) [reply src/use_case/collect_recipe/CollectRecipeInputData.java:3]: Hi, I do not see where the "opened folder view model" you are referring to. I did implement collect button should in RecipeView

File: src/data_access/FileFolderDataAccessObject.java (hunk 1 above)
Comment (shiqio3o) [reply src/data_access/FileFolderDataAccessObject.java]: yes  we can use yours now

**Review by yuxizzz — COMMENTED** [review]
ok

**Pull request merged**